### ISSUE TITLE
RBM : Shuffle database

### ISSUE THREAD

**Issue by FrancoisLasson** [FrancoisLasson/Temporal_DBN#3]
This issue answers #1 

As we say in #1, we want to check the influence of a shuffle dataset on the cost evolution. 
In fact,  RBM is trained one by one using gesture after another. This principle
could create a bias, the last gesture could impact quiet more the
weights of the RBM than the other gestures. In order to delete this
prospective bias, we shuffle the database. So, all gestures are learned
in the same time and have the same influence on the RBM’s weights.

To check this influence, we have trained RBMs in the same conditions : 
number of epochs = 10000, learning rate = 1e-3, number of hidden neurones = 30, batch_size=100

•Training,  gesture ather another : 
![rbm_1459954702 43](https://cloud.githubusercontent.com/assets/11289891/14321693/8a10716e-fc1a-11e5-9254-3cca8194b11a.png)

•With shuffle : 
![rbm_1459943933 76-shuffle](https://cloud.githubusercontent.com/assets/11289891/14315910/4d6daa0c-fc00-11e5-80f0-95e5e5aff4d1.png)

As we could could see, shuffle the dataset decrease costs amplitude of the RBM4 and avoid the oscillation effects of the RBM0. Even if it has a quiet negative impact on other RBMs, on the whole **shuffle dataset improve the learning phase**
